pkg/api/persistentvolume/util.go

```diff
@@ -33,12 +33,5 @@ func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.Persist
 }
 
 func hasExpansionSecrets(oldPVSpec *api.PersistentVolumeSpec) bool {
-	if oldPVSpec == nil || oldPVSpec.CSI == nil {
-		return false
-	}
-
-	if oldPVSpec.CSI.ControllerExpandSecretRef != nil {
```

Comment:
Do these `if` statements represent different `concepts`? If yes, I think I'm in favor of keeping them separate. However, I think we can get rid of the `return false` in favor of `return oldPVSpec...`

I think my opinions here apply to much of the `if` statements in this pr. In general, I support separate if statements for separate concepts, and avoiding `return (true|false)` when its possible to return the results of an expression instead.